src/test/java/uk/gov/register/functional/AnalyticsFunctionalTest.java

```diff
@@ -0,0 +1,121 @@
+import uk.gov.register.functional.app.WipeDatabaseRule;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.Assert.assertThat;
+import static uk.gov.register.functional.db.TestDBSupport.postgresConnectionString;
+
+@RunWith(Parameterized.class)
+public class AnalyticsFunctionalTest {
+
+    private static final String TRACKING_ID_NOT_PRESENT = null;
+    private static final String TRACKING_ID_EMPTY = "";
+    private static final String TRACKING_ID_VALID = "UA-12345678-1";
+
+    @Rule
+    public TestRule wipe = new WipeDatabaseRule();
+
+    @Rule
+    public DropwizardAppRule<RegisterConfiguration> appRule;
+
+    private final String targetUrl;
+    private final String trackingId;
+    private final Boolean shouldIncludeAnalytics;
+
+
+    public AnalyticsFunctionalTest(String targetUrl, String trackingId, Boolean shouldIncludeAnalytics) {
+        this.appRule = createAppRule(trackingId);
+        this.targetUrl = targetUrl;
+        this.trackingId = trackingId;
+        this.shouldIncludeAnalytics = shouldIncludeAnalytics;
+    }
+
+    @Parameterized.Parameters(name = "{index} - url: {0} with code: {1} should include analytics: {2}")
+    public static List data() {
+        List sourceData = new ArrayList();
+
+        sourceData.addAll(generateTestSetFor("/"));
+        sourceData.addAll(generateTestSetFor("/download"));
+        sourceData.addAll(generateTestSetFor("/entries"));
+        sourceData.addAll(generateTestSetFor("/records"));
+        sourceData.addAll(generateTestSetFor("/not-found-page"));
+
+        return sourceData;
+    }
+
+    @Test
+    public void checkAnalyticsScriptsPresence() throws Exception {
+        Client client = getTestClient(this.appRule);
+        Response response = client.target(String.format("http://localhost:%d%s", this.appRule.getLocalPort(), targetUrl))
+                .request().get();
+
+        Document doc = Jsoup.parse(response.readEntity(String.class));
+        assertThat(response.getStatus(), lessThan(500));
+
+        Element trackingIdElem = doc.getElementById("analytics-tracking-id");
+        Boolean docIncludesAnalyticsId = trackingIdElem != null;
+        Boolean docIncludesMainAnalytics = doc.getElementById("analytics-main") != null;
+        Boolean docIncludesExtLinksAnalytics = doc.getElementById("analytics-external-links") != null;
+
+        assertThat(docIncludesAnalyticsId, equalTo(shouldIncludeAnalytics));
+        assertThat(docIncludesMainAnalytics, equalTo(shouldIncludeAnalytics));
+        assertThat(docIncludesExtLinksAnalytics, equalTo(shouldIncludeAnalytics));
+
+        if (shouldIncludeAnalytics) {
+            assertThat(docIncludesAnalyticsId, equalTo(true));
```

Comment:
was added for clarity of intention, will remove it if the test is clear without it